src/goto-harness/function_call_harness_generator.cpp

```diff
@@ -144,6 +206,17 @@ void function_call_harness_generatort::implt::generate(
   auto const arguments = declare_arguments(function_body);
 
   recursive_initialization_config.mode = function_to_call.mode;
+  recursive_initialization_config.pointers_to_treat_as_arrays =
```

Comment:
This part is getting a bit more involved for one function than I am comfortable with, may be worth splitting this up into another function

Reply:
We decided to leave it as is right now, but to add an extra comment for slightly better documentation.